src/components/HyloEditor/HyloEditor.js

```diff
@@ -81,9 +84,19 @@ export default class HyloEditor extends Component {
   }
 
   componentDidUpdate (prevProps) {
+    // regenerate state from content if HTML changes
     if (this.props.contentHTML !== prevProps.contentHTML) {
       this.setState(this.defaultState(this.props))
     }
+
+    // handle initial focus behaviour
+    if (this.props.focusOnRender !== prevProps.focusOnRender) {
+      this.setState({ didInitialFocus: false })
+    }
+    if (this.props.focusOnRender && !this.state.didInitialFocus) {
+      this.focus()
+      this.setState({ didInitialFocus: true })
```

Comment:
this doesnt quite seem right to me, you could do two re-renders here with two setState calls, seems like you should check to only do one between both these conditions?

Reply:
Nice catch! Let me know if da5c6b97 doesn't resolve that for you.